### ISSUE TITLE
Rasplex not responhding to key press via CEC

### ISSUE THREAD

**Issue by norbusan** [RasPlex/RasPlex#378]
I am having the following setup: Rasplex 0.6.0 on Raspberry II, connected via HDMI to Sony TV. Sony's BraviaLink is turned on, and sends key press from the remote to rasplex.

I can see the key press even in the log files:
`````
08:29:57 T:1866462288   DEBUG: CecLogMessage - >> Free use (E) -> Recorder 1 (1): user control pressed (44)
08:29:57 T:1866462288   DEBUG: CecLogMessage - SetCurrentButton down (2) D:0ms cur:2
08:29:57 T:1866462288   DEBUG: CecLogMessage - key pressed: down (2) current(ff) duration(0)
08:29:57 T:1866462288   DEBUG: CecLogMessage - Changed key down (2) D:0ms cur:ff
08:29:57 T:1866462288   DEBUG: CecLogMessage - key pressed: down (2, 0)
08:29:57 T:1866462288   DEBUG: PushCecKeypress - received key a7 duration 0
08:29:57 T:1866462288   DEBUG: CecLogMessage - CheckKeypressTimeout T:299.269
08:29:57 T:1866462288   DEBUG: CecLogMessage - Key down: idle (duration:0) (2) timeout:500ms (rel:500,rep:0,prs:1,rel:0)
08:29:57 T:1866462288   DEBUG: CecLogMessage - >> e1:8b:02
08:29:57 T:1866462288   DEBUG: CecLogMessage - >> Free use (E) -> Recorder 1 (1): vendor remote button up (8B)
08:29:57 T:1866462288   DEBUG: CecLogMessage - key released: down (2) D:218ms
08:29:57 T:1866462288   DEBUG: PushCecKeypress - received key a7 duration 218
08:29:58 T:1799353424   DEBUG: CecLogMessage - << Recorder 1 (1) -> TV (0): POLL
08:29:58 T:1799353424   DEBUG: CecLogMessage - << 10
08:29:58 T:1799353424   DEBUG: Previous line repeats 1 times.
08:29:58 T:1799353424   DEBUG: CecLogMessage - >> POLL not sent
`````
but there is no reaction from the PHT/Rasplex GUI at all.

I tried resetting my storage area, without change.

Thanks

Norbert

**Comment by NedtheNerd** [comment]
Can I suggest that you try refreshing the SD card.

**Comment by norbusan** [comment]
Tried that now, same effect. Grabed the 0.6.0 release for RP2, flashed it to my SD card, installed it into the RP2, rasplex booted, enlarged the fs, rebooted, usual setup procedure.

Then I actived debug logging and restarted rasplex, logged per ssh into the device, tail -f /storage/.plexht/temp/plexhometheater.log and I get similar events as above:
`````
15:34:26 T:1858073680   DEBUG: CecLogMessage - >> Free use (E) -> Recorder 1 (1): user control pressed (44)
15:34:26 T:1858073680   DEBUG: CecLogMessage - SetCurrentButton down (2) D:0ms cur:2
15:34:26 T:1858073680   DEBUG: CecLogMessage - key pressed: down (2) current(ff) duration(0)
15:34:26 T:1858073680   DEBUG: CecLogMessage - Changed key down (2) D:0ms cur:ff
15:34:26 T:1858073680   DEBUG: CecLogMessage - key pressed: down (2, 0)
15:34:26 T:1858073680   DEBUG: PushCecKeypress - received key a7 duration 0
15:34:26 T:1858073680   DEBUG: CecLogMessage - CheckKeypressTimeout T:161.300
15:34:26 T:1858073680   DEBUG: CecLogMessage - Key down: idle (duration:0) (2) timeout:500ms (rel:500,rep:0,prs:1,rel:0)
15:34:27 T:1858073680   DEBUG: CecLogMessage - >> e1:8b:02
15:34:27 T:1858073680   DEBUG: CecLogMessage - >> Free use (E) -> Recorder 1 (1): vendor remote button up (8B)
15:34:27 T:1858073680   DEBUG: CecLogMessage - key released: down (2) D:211ms
15:34:27 T:1858073680   DEBUG: PushCecKeypress - received key a7 duration 211
15:34:27 T:1799095376   DEBUG: CecLogMessage - expected response not received (90: report power status)
15:34:27 T:1799095376   DEBUG: CecLogMessage - << 10:8f
15:34:27 T:1923085392   DEBUG: Error: Requested setting (videoscreen.textures32) was not found.  It must be case-sensitive
`````

Anything else I can try? It looks strange that the Cec logs receiving a key press, but the rasplex doesn't react?

**Comment by norbusan** [comment]
BTW, is there any chance that dropping something into `/storage/.plexht/userdata/keymaps/` might change something?

**Comment by NedtheNerd** [comment]
Yes, it definitely will, any xml file in their will overwrite the default and if incorrect will stop CEC working, remove the file, reboot and see if this issue still presents itself.

**Comment by norbusan** [comment]
Hi
sorry for the misunderstanding, there is *no* file there, I thought by dropping one in there it might go back to work ;-)

**Comment by NedtheNerd** [comment]
I have a Sony Bravia TV and CEC works out of the box, check that CEC is enabled.

**Comment by norbusan** [comment]
Hmm, it looks like enabled, otherwise the keypresses would not *arrive* in rasplex, or??? Why are they mentioned in the rasplex log file? That means the CEC is connected and enabled, as far as I understand.

**Comment by NedtheNerd** [comment]
It just means that Rasplex is confirming that a CEC signal has been generated and is being sent to the TV, for some reason your TV is not either receiving the signal or is not processing it.  That fact that you have the entry in the log file proves that this is not a Rasplex issue.  Have you checked the Bravia Sync settings on your TV.  Try swapping the HDMI cable, some are not that good, especially ones that are less than HDMI 1.4 spec. If this really was a Rasplex issue the forums would be full of complaints.

Now closing issue.

**Comment by norbusan** [comment]
Hmmm, I don't understand your answer. Why should the *TV* react?

My interpretations is:
I press button on remote -> TV receives the signal -> TV forwards the signal via HDMI to rasplex -> Rasplex receives the signal -> Rasplex reacts (scroll up, down etc).

Now, I can confirm that the information is *received* on the rasplex side, as clearly shown on in the log files. So there is nothing more the TV is supposed to do. Rasplex needs to act on the pressed *and* received buttons.
